libs/wingsdk/src/target-tf-gcp/function.ts

```diff
@@ -108,6 +116,49 @@ export class Function extends cloud.Function {
     );
   }
 
+  public addBucketPermission(
+    bucket: Bucket,
+    permission: StorageBucketPermissions
+  ): void {
+    const app = App.of(this) as App;
+
+    // check if the permission is already set
+    if (this.bucketPermission.has(`${bucket.bucket.name}-${permission}`)) {
+      return;
+    }
+
+    // as gcp bucket has unique name, we can use bucket name in map as unique id along with permission
+    if (permission === StorageBucketPermissions.READ) {
+      let newReadPermission = new StorageBucketIamMember(
+        this,
+        "BucketPermissionToRead",
+        {
+          bucket: bucket.bucket.name,
+          role: "roles/storage.objectViewer",
+          member: `projectViewer:${app.projectId}`,
+        }
+      );
+      this.bucketPermission.set(
+        `${bucket.bucket.name}-${StorageBucketPermissions.READ}`,
+        newReadPermission
+      );
```

Comment:
nit: can add early return to remove `else if`

Reply:
Are you suggesting this? 
```js
if(permission!==StorageBucketPermissions.READ && permission!==StorageBucketPermissions.READWRITE){
      throw new Error("Invalid permission");
      }
```
Then it would be only else
```js
else {
      let newReadWritePermission = new StorageBucketIamMember(
        this,
        "BucketPermissionToWrite",
        {
          bucket: bucket.bucket.name,
          role: "roles/storage.objectAdmin",
          member: `projectEditor:${app.projectId}`,
        }
      );
      this.bucketPermission.set(
        `${bucket.bucket.name}-${StorageBucketPermissions.READWRITE}`,
        newReadWritePermission
      );
    }
```

Reply:
yes, I also just noticed. instead of enum, I did this. 
```js
if (permission === StorageBucketPermissions.READ) {
      let newReadPermission = new StorageBucketIamMember(
        this,
        "BucketPermissionToRead",
        {
          bucket: bucket.bucket.name,
          role: permission,
          member: `projectViewer:${app.projectId}`,
        }
      );
      this.bucketPermission.set(
        `${bucket.bucket.name}-${permission}`,
        newReadPermission
      );
    }
```
I directly used the permissions props.

Reply:
Great! :)